Tests/SentryTests/SentryCrash/SentryCrashDynamicLinkerTests.m

```diff
@@ -0,0 +1,154 @@
+    XCTAssertEqual(info.dli_saddr, dyldAddress, @"Symbol address should match the input address");
+}
+
+#if TARGET_OS_IOS
+- (void)testUIKitAddressLookup
+{
+    // Get a known function from UIKit
+    void *uiKitAddress = (void *)&UIApplicationMain;
+
+    Dl_info info;
+    bool result = sentrycrashdl_dladdr((uintptr_t)uiKitAddress, &info);
+    XCTAssertTrue(result, @"dladdr should succeed for UIKit address");
+    XCTAssert(info.dli_fbase != NULL, @"Base address should not be NULL");
+    XCTAssert(info.dli_fname != NULL, @"Image name should not be NULL");
+
+    XCTAssertTrue(strstr(info.dli_fname, "UIKit") != NULL, @"Image name should contain 'UIKit'");
+
+    XCTAssert(info.dli_sname != NULL, @"Symbol name should not be NULL");
+    XCTAssertTrue(strstr(info.dli_sname, "UIApplicationMain") != NULL,
+        @"Symbol name should contain 'UIApplicationMain'");
+
+    XCTAssert(info.dli_saddr != NULL, @"Symbol address should not be NULL");
+    XCTAssertEqual(info.dli_saddr, uiKitAddress, @"Symbol address should match the input address");
+}
+#endif
+
+- (void)testKnownAddressLookup
+{
+    // Any function in Sentry will do
+    void *testAddress = (void *)&sentrycrashdl_clearDyld;
+
+    Dl_info info;
+    bool result = sentrycrashdl_dladdr((uintptr_t)testAddress, &info);
+    XCTAssertTrue(result, @"dladdr should succeed for test bundle address");
+    XCTAssert(info.dli_fbase != NULL, @"Base address should not be NULL");
+    XCTAssert(info.dli_fname != NULL, @"Image name should not be NULL");
+
+    XCTAssertTrue(
+        strstr(info.dli_fname, "Sentry") != NULL, @"Image name should contain 'SentryTests'");
+
+    XCTAssert(info.dli_sname != NULL, @"Symbol name should not be NULL");
+    XCTAssertTrue(strstr(info.dli_sname, "sentrycrashdl_clearDyld") != NULL,
+        @"Symbol name should contain 'sentrycrashdl_clearDyld'");
+
+    XCTAssert(info.dli_saddr != NULL, @"Symbol address should not be NULL");
+    XCTAssertEqual(info.dli_saddr, testAddress, @"Symbol address should match the input address");
+}
+
+// vmaddrs changes by platform, so we cannot use a static value
+- (uintptr_t)findDyldAddress
+{
+    struct dyld_all_image_infos *infos = getAllImageInfo();
+    const struct mach_header *header = NULL;
+    if (infos && infos->dyldImageLoadAddress) {
+        header = (const struct mach_header *)infos->dyldImageLoadAddress;
+    }
+
+    // Calculate an address within dyld TEXT segment (varies by platform)
+    uintptr_t cmdPtr = firstCmdAfterHeader(header);
+    for (uint32_t iCmd = 0; iCmd < header->ncmds; iCmd++) {
+        const struct load_command *loadCmd = (struct load_command *)cmdPtr;
+        if (loadCmd->cmd == SEGMENT_TYPE) {
+            const segment_command_t *segCmd = (segment_command_t *)cmdPtr;
+            if (strcmp(segCmd->segname, "__TEXT") == 0) {
+                return (uintptr_t)header + segCmd->vmaddr;
+            }
+        }
+        cmdPtr += loadCmd->cmdsize;
+    }
+    return (uintptr_t)header;
+}
```

Comment:
This looks like a combination of two separate pieces of implementation from the SDK, any chance we can just expose the function through a +Test.h header?